snap/validate_test.go

```diff
@@ -103,9 +103,9 @@ func (s *ValidateSuite) TestValidateVersion(c *C) {
 		// can't have non-whitelisted symbols
 		"v1.3([<$$$>])", "what even _is_ a version",
 		// can't start with whitelisted symbls
-		":", ".", "+", "~", "_", "-",
+		":", ".", "+", "~", "-",
```

Comment:
Let's add a version with an underscore here, it will be better than not having any tests.